cmd/controller/main.go

```diff
@@ -282,7 +287,9 @@ func parseEnvFlags() config {
 	pflag.Bool(pullSidecarFlag, viper.GetBool(pullSidecarFlag), "For development purposes, set the sidecar image to have a ImagePullPolicy of Always. Can also use ALWAYS_PULL_SIDECAR env variable")
 	pflag.String(sdkServerAccountFlag, viper.GetString(sdkServerAccountFlag), "Overwrite what service account default for GameServer Pods. Defaults to Can also use SDK_SERVICE_ACCOUNT")
 	pflag.Int32(minPortFlag, 0, "Required. The minimum port that that a GameServer can be allocated to. Can also use MIN_PORT env variable.")
-	pflag.Int32(maxPortFlag, 0, "Required. The maximum port that that a GameServer can be allocated to. Can also use MAX_PORT env variable")
+	pflag.Int32(maxPortFlag, 0, "Required. The maximum port that that a GameServer can be allocated to. Can also use MAX_PORT env variable.")
+	pflag.StringToInt(portRangeMinPortFlag, cast.ToStringMapInt(viper.GetStringMap(portRangeMinPortFlag)), "Optional. List of port range min ports. Can Also use PORT_RANGE_MIN_PORT env variable.")
```

Comment:
Out of curiosity (or maybe also docs?) what's the format here for the list?

Reply:
I had originally thought from a perusal of the code it was the standard `key=value`, but in reality it is a JSON object. I don't love it.

Reply:
I have reworded to include the format in the description.

Reply:
I don't think this should be two concurrent flags. I would prefer a single flag that took a structure JSON blob as config (e.g. as I mentioned here: https://github.com/googleforgames/agones/pull/3747#discussion_r1557890594), it's a little weird to take two?

Reply:
I'm not necessarily saying to force the JSON all the way from the helm - doesn't gotemplate support building maps? you can rebuild the structure in the helm template, not dissimilar to what you're doing today. And what I mean is: I like the configuration in helm, but I think the helm template could translate to JSON. It's fussy but I think it's doable.

Reply:
When I say configure from YAML or TOML, I mean config from a file, not from helm. This is the primary use of `viper` is to allow CLI, Env and File config.